packages/core/test/template-engine/built-in-extensions/cache-layer/cache.spec.ts

```diff
@@ -0,0 +1,162 @@
+import { arrayToStream, streamToArray } from '@vulcan-sql/core/utils';
+import { createTestCompiler } from '../../testCompiler';
+
+it('Extension compiled succeed when exist symbol "variable" after "cache" tag', async () => {
+  // Arrange
+  const { compiler } = await createTestCompiler();
+
+  // Action, Assert
+  await expect(
+    compiler.compile(`{% cache variable %} some statement {% endcache %}`)
+  ).resolves.not.toThrow();
+});
+
+it('Extension compiled succeed when the token is end block after "cache" tag', async () => {
+  // Arrange
+  const { compiler } = await createTestCompiler();
+
+  // Action, Assert
+  await expect(
+    compiler.compile(`{% cache %} some statement {% endcache %}`)
+  ).resolves.not.toThrow();
+});
+
+it('If argument have too many elements, extension should throw an error', async () => {
+  // Arrange
+  const { compiler } = await createTestCompiler();
+
+  // Action, Assert
+  await expect(
+    compiler.compile(`
+{% cache variable more %}
+select count(*) as count from user where user.id = '{{ params.userId }}';
+{% endcache %}
+  `)
+  ).rejects.toThrow(`Expected a block end, but got symbol`);
+});
+
+it('If argument is not a symbol or end block, extension should throw', async () => {
```

Comment:
After discussion, because the Extension not have the compile method, and it delegates the compiler to parse in the compiling phase, so we change to use `should be compiled failed` to the suitable description